lib/models/ui.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const UI = require('console-ui');
+
+let singletonUI;
+
+/**
+ * The class that serves as a singleton wrapper for `console-ui`.
+ *
+ * @private
+ * @module ember-cli
+ * @class CommandLineUI
+ * @constructor
+ */
+class CommandLineUI {
+  constructor(args) {
+    let options = args || {};
+
+    if (!singletonUI) {
+      singletonUI = new UI({
+        inputStream: options.inputStream,
```

Comment:
we use `closestSync` only in one place - [`lib/models/project`](https://github.com/ember-cli/ember-cli/search?utf8=%E2%9C%93&q=closestSync) and we do not pass in `ui` or `cli` parameters when we call it.

the reason the fallback was put in place was b/c we created 2 instances of the `UI`: one when `bin/ember` runs and the other one when we require `lib/broccoli/ember-app`.

with this change, `UI` is created only once and we no longer require this fallback.

perhaps I'm missing something obvious?